src/main/native/compression/IntelDeflater.cc

```diff
@@ -157,30 +166,44 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_compression_IntelDeflater_resetNative
 
 JNIEXPORT void JNICALL Java_com_intel_gkl_compression_IntelDeflater_generateHuffman
 (JNIEnv * env, jobject obj) {
-
-      jbyteArray inputBuffer = (jbyteArray)env->GetObjectField(obj, FID_inputBuffer);
-
       jint level = env->GetIntField(obj, FID_level);
 
       if(level == 1) {
       isal_zstream* lz_stream = (isal_zstream*)env->GetLongField(obj, FID_lz_stream);
+      if (lz_stream == NULL) {
+        if(env->ExceptionCheck())
+            env->ExceptionClear();
+        env->ThrowNew(env->FindClass("java/lang/NullPointerException"), "lz_stream is NULL.");
+        return;
+      }
 
+      jbyteArray inputBuffer = (jbyteArray)env->GetObjectField(obj, FID_inputBuffer);
       jbyte* input = (jbyte*)env->GetPrimitiveArrayCritical(inputBuffer, 0);
 
+      if (input == NULL) {
+        if (env->ExceptionCheck())
+          env->ExceptionClear();
+        env->ThrowNew(env->FindClass("java/lang/NullPointerException"), "inputBuffer is null.");
+        env->ReleasePrimitiveArrayCritical(inputBuffer, input, 0);
+        return;
+      }
+
       struct isal_huff_histogram *histogram = (struct isal_huff_histogram *) malloc(sizeof(*histogram)); 
       struct isal_hufftables *hufftables_custom;
 
       if (histogram == NULL) {
-           DBG ("Malloc failed out of memory");
-           return; 
-      } 
-      
+        DBG ("Malloc failed out of memory");
+        if(env->ExceptionCheck())
+            env->ExceptionClear();
+        env->ThrowNew(env->FindClass("java/lang/OutOfMemoryError"),"Memory allocation error");
+        return;
+      }
+
       memset(histogram, 0, sizeof(isal_huff_histogram));
       isal_update_histogram((unsigned char*)input, 64*1024, histogram);
       isal_create_hufftables(hufftables_custom, histogram);
       lz_stream->hufftables = hufftables_custom;
 
-      env->SetLongField(obj, FID_lz_stream, (jlong)lz_stream);
```

Comment:
Where did this "Set" go?

Reply:
This was unnecessary as the pointer value never gets modified in the course of the method.